### PR TITLE
Automate generation of antrea manifest template

### PR DESCRIPTION
**Pull request by ksamoray**
antrea-manifest/antrea.yml is a template file which is based on contents
from Antrea repository.

This template file should be in sync with the contents in the antrea
repo, and this PR automates the synchronization of the file with the
Antrea repo.

To generate the manifest, use hack/generate-manifest.sh
Alternately use make antrea-manifest target to build the manifest and
put it in place into antrea-manifest dir.

Signed-off-by: Kobi Samoray <ksamoray@vmware.com>

### CODE REVIEW

File: hack/verify-kustomize.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+verify_kustomize() {
+    # Check if there is already a kustomize binary in $_BINDIR and if yes, check
+    # if the version matches the expected one.
+    local kustomize="$(PATH=$_BINDIR command -v kustomize)"
+    if [ -x "$kustomize" ]; then
+        # Verify version if kustomize was already installed.
+        local kustomize_version="$($kustomize version --short)"
+        # Should work with:
+        #  - kustomize/v3.3.0
+        #  - {kustomize/v3.8.2  2020-08-29T17:44:01Z  }
```
Comment (salv-orlando) [inline hack/verify-kustomize.sh:33]: I know this is the same as Antrea, but the comment does not have much sense if we're now using kustomize 4.4.1...

File: hack/generate-manifest.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+if [ -z "$KUSTOMIZE" ]; then
+    KUSTOMIZE="$(verify_kustomize)"
+elif ! $KUSTOMIZE version > /dev/null 2>&1; then
+    echoerr "$KUSTOMIZE does not appear to be a valid kustomize binary"
+    print_help
+    exit 1
+fi
+
+# Get Antrea repository
+ANTREA_DIR=$(mktemp -d /tmp/antrea.XXXXXXX)
```
Comment (salv-orlando) [inline hack/generate-manifest.sh:64]: out of curiosity, is there a logic for the suffix length or are we just typing 'X' until we've had enough?

File: hack/generate-manifest.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+# unfortunately 'kustomize edit add configmap' does not support specifying 'merge' as the behavior,
+# which is why we use a template kustomization file.
+sed -e "s/<AGENT_CONF_FILE>/antrea-agent.conf/; s/<CONTROLLER_CONF_FILE>/antrea-controller.conf/; s/<CNI_CONFLIST_FILE>/antrea-cni.conflist/" $THIS_DIR/../build/yamls/patches/templates/kustomization.configMap.tpl.yml > kustomization.yml
+$KUSTOMIZE edit add base $BASE
+BASE=../configMap
+cd ..
+
+mkdir osmft && cd osmft
+cp $THIS_DIR/../build/yamls/patches/*.yml .
+touch kustomization.yml
```
Comment (salv-orlando) [inline hack/generate-manifest.sh:100]: Do you know why we need to call 'touch' here and below?

File: hack/generate-manifest.sh [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+$KUSTOMIZE edit add base $BASE
+$KUSTOMIZE edit add patch --path agentOsRelease.yml
```
Comment (salv-orlando) [inline hack/generate-manifest.sh:102]: pedant nit: the acronym for openshift is usually OCP, to distinguish it from OS which is often used for Openstack.

File: hack/verify-kustomize.sh (hunk 1 above)
Comment (ksamoray) [reply hack/verify-kustomize.sh:33]: Yep it's redundant. I'll remove the comment.

File: hack/generate-manifest.sh (hunk 2 above)
Comment (ksamoray) [reply hack/generate-manifest.sh:64]: mktemp will generate a file (or a dir in this case as -d is specified) with a name that matches the template. 'X'es are replaced with random characters - the length of the X string determines the length of the random string.

File: hack/generate-manifest.sh (hunk 3 above)
Comment (ksamoray) [reply hack/generate-manifest.sh:100]: To create an emptry kustomization.yml file

File: hack/generate-manifest.sh (hunk 4 above)
Comment (ksamoray) [reply hack/generate-manifest.sh:102]: Renamed, thanks!

**Review by salv-orlando — DISMISSED** [review]

**Comment by ksamoray** [comment]
@salv-orlando as the change is approved, I'm gonna regenerate antrea-manifest/antrea.yml and include it within the PR.

**Review by salv-orlando — APPROVED** [review]

**Pull request merged**